runtime/vm/createramclass.cpp

```diff
@@ -1750,6 +1750,10 @@ loadFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *class
 						goto done;
 					}
 
+					if (!J9_IS_J9CLASS_FLATTENED(valueClass)) {
+						valueClass->classFlags |= J9ClassContainsUnflattenedFlattenables;
```

Comment:
This is not the correct class to set. You need to set this flag on the class that is currently being created.

`largestFieldType` is being used to return flags, you can rename this to `valueTypeFlags` and set it.